### ISSUE TITLE
String not converted to regexp

### ISSUE THREAD

**Issue by rovf** [seamusabshere/to_regexp#3]
`irb(main):001:0> require 'to_regexp'
C:/cygwin64/home/FISRONA/gitwrk/vp5/gem/gems/to_regexp-0.2.1/lib/to_regexp.rb:35 warning: `*' interpreted as argument prefix
true
....
irb(main):008:0>  'a.b'.as_regexp(:literal => false)
nil
irb(main):010:0>
`

Since I have set the :literal option explicitly, it should turn 'a.b' into /a.b/, even though it doesn't have delimiters.

**Comment by seamusabshere** [comment]
@rovf 

```
irb(main):004:0> 'a.s'.to_regexp
=> nil
irb(main):005:0> 'a.s'.to_regexp(literal: true)
=> /a\.s/
```

is it possible that `literal` means the opposite of what you wanted?

**Comment by rovf** [comment]
> irb(main):005:0> 'a.s'.to_regexp(literal: true)

> is it possible that `literal` means the opposite of what you wanted?

It is a typo. It should be

   'a.s'.to_regexp(literal: false)

and this would incorrectly return nil. I will fix the text of the issue.
Thank you for pointing this out.

Ronald

**Comment by rovf** [comment]
No, I did _not_ want literal interpretation. I want the content to be interpreted as regexp, but without supplying delimiters. 

I set literal: false, because if I would not mention the "literal" option, to_regexp should indeed return nil (according to the documentation, respective commentary).

Basically, the effect I had expected, would be the equivalent to

  Regexp.new('a.s')

**Comment by joshco** [comment]
Just came across this cool gem and I see @rovf 's point.  My use case for seeking this gem is to allow user configurable querying via regexps contained in an active record model.  I need an easy way to pull the string from the db column and turn it into a regexp, without having to worry about if the user put in the surrounding delimiters. So I'd like the delimiters to be added if not present, but not treat the rest of the string as literals.

So if my user has lamely entered fo.bar  I want it to be /fo.bar/ not /foo.bar/

**Comment by rovf** [comment]
@joshco: I don't think the "delimiters should be added if not present" would be a good idea. Imagine the user wants to be his regexp pattern a forward slash, followed by an arbitrary number of other characters, followed by another forward slash. If he wrote it as /._/, the algorithm you propose would interpret the / as delimiters, and use only ._ for the pattern. In fact, the user would be required to enter the pattern as [/].*/ - not very intuitive.

In fact, in your example, delimiters should never be assumed. The purpose of a regexp delimiter is to clearly denote start and end of a regexp within a program (a regexp is a "language used within a language"). If the regexp is stored in a database, there is never a need for a delimiter, because the regexp is not embedded in something else.
